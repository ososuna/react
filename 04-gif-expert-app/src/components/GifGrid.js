
import React, { useState, useEffect } from 'react'
import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);
    
    useEffect( async() =>{
        const images = await getGifs( category );
        setImages( images );
    }, [ category ]);
    
    return (
        <>
            <h3>{ category }</h3>
            <div className="card-grid">
                    {
                        images.map( ( img ) => (
                            <GifGridItem
                                key={ img.id }
                                { ...img }
                            />
                        ))
                    }
            </div>
        </>
    )
}
