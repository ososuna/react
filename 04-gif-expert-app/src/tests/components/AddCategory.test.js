
import { React } from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import { AddCategory } from '../../components/AddCategory';

describe('AddCategory component tests', () => {
    
    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={ setCategories } /> );

    beforeEach( () => {
       jest.clearAllMocks();
       wrapper = shallow( <AddCategory setCategories={ setCategories } /> );
    });

    test('should be shown correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should change the text input', () => {
        
        const input = wrapper.find('input');
        const value = 'Hello World';
        
        input.simulate('change', { target: { value } });
    
    });

    test('should not post on submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} });    
        expect( setCategories ).not.toHaveBeenCalled();
    });
    
    test('should call setCateggories and clean the text input', () => {
        
        const input = wrapper.find('input');
        const value = 'Hello World';
        
        input.simulate('change', { target: { value } });

        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledWith( expect.any( Function ) );

        expect( input.prop('value') ).toBe('');

    });
    
});
