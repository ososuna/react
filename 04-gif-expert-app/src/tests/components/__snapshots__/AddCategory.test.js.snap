// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory component tests should be shown correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
